### PR TITLE
test: add upgrades tests to on-chain-relayer contracts

### PR DESCRIPTION
**Pull request by koonopek**

### CODE REVIEW

File: packages/on-chain-relayer/test/contracts/price-feeds/common/price-feed-utils.ts [hunk 1]
```diff
@@ -127,12 +129,66 @@ export const describeCommonPriceFeedTests = ({
...
     it("should properly upgrade the contract (change data feed adapter)", async () => {
-      expect(1).to.be.equal(1);
+      const updateContractFactory = await ethers.getContractFactory(
+        "PriceFeedsUpdatedMock"
+      );
+
+      const updatedContract = (await upgrades.upgradeProxy(
+        contractV1,
+        updateContractFactory
+      )) as PriceFeedBase;
```
Comment (cehali) [inline packages/on-chain-relayer/test/contracts/price-feeds/common/price-feed-utils.ts:168]: This part is the same here and in `should properly upgrade the contract (change data feed id)` test. It seems that in both cases you upgrade the price feed contract

Comment (cehali) [reply packages/on-chain-relayer/test/contracts/price-feeds/common/price-feed-utils.ts:168]: ohhh, my bad sorry

Comment (cehali) [reply packages/on-chain-relayer/test/contracts/price-feeds/common/price-feed-utils.ts:168]: But still, this part is the same, maybe put it in one test?

File: packages/on-chain-relayer/contracts/mocks/PriceFeedsUpdatedMock.sol [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+pragma solidity ^0.8.4;
+
+import "../price-feeds/with-rounds/PriceFeedWithRounds.sol";
+
+contract PriceFeedsUpdatedMock is PriceFeedWithRounds {
```
Comment (hatskier) [inline packages/on-chain-relayer/contracts/mocks/PriceFeedsUpdatedMock.sol:7]: PriceFeedsUpdatedMock -> PriceFeedUpdatedMock?

File: packages/on-chain-relayer/test/contracts/price-feeds/common/price-feeds-adapter-utils.ts [hunk 3]
```diff
@@ -104,20 +104,68 @@ export const describeCommonPriceFeedsAdapterTests = ({
...
+
+      expect(dataFeeds.length).to.eq(2);
+      expect(dataFeeds).to.deep.eq([
+        "0x4554480000000000000000000000000000000000000000000000000000000000",
+        "0x4254430000000000000000000000000000000000000000000000000000000000",
+      ]);
+    });
+
+    it("should properly upgrade the contract (change authorized updaters)", async () => {
+      const updatedContractFactory = await ethers.getContractFactory(
```
Comment (hatskier) [inline packages/on-chain-relayer/test/contracts/price-feeds/common/price-feeds-adapter-utils.ts:154]: Can we check another authorised signer before the contract update to be sure that it has changed?

**Review by hatskier — COMMENTED** [review]
Can you please also implement tests for access control for the upgrades? We need to be able to specify who is the "owner" (who can update the contracts)

File: packages/on-chain-relayer/test/contracts/price-feeds/common/price-feed-utils.ts (hunk 1 above)
Comment (koonopek) [reply packages/on-chain-relayer/test/contracts/price-feeds/common/price-feed-utils.ts:168]: Good point
Refactored common part to new describe block `before`

File: packages/on-chain-relayer/test/contracts/price-feeds/common/price-feeds-adapter-utils.ts (hunk 3 above)
Comment (koonopek) [reply packages/on-chain-relayer/test/contracts/price-feeds/common/price-feeds-adapter-utils.ts:154]: will do

**Comment by koonopek** [comment]
> Can you please also implement tests for access control for the upgrades? We need to be able to specify who is the "owner" (who can update the contracts)

Hmm but where such tests should be placed, because it is more like testing `ProxyAdmin`, `TransparentProxy` and plugin by openzeppelin then our contracts?

**Review by hatskier — APPROVED** [review]

**Pull request merged**